crates/assert-instr-macro/src/lib.rs

```diff
@@ -133,6 +134,8 @@ pub fn assert_instr(
         fn #assert_name() {
             #to_test
 
+            unsafe { asm!("" : : "r"(#shim_name as usize) : "memory" : "volatile") };
```

Comment:
Before we never called the shim for anything, we passed a pointer to it to `stdsimd_test::assert`, but AFAICT that function never called that pointer either (maybe that's what `backtrace` was doing?). 

Either way, we leak the address of this pointer to "all memory" here, and perform arbitrary side-effects, which could include calling this function. So with this LLVM should never be able to remove the function.